crates/ra_hir_def/src/import_map.rs

```diff
@@ -45,13 +47,17 @@ pub struct ImportMap {
     /// the index of the first one.
     importables: Vec<ItemInNs>,
     fst: fst::Map<Vec<u8>>,
+
+    /// Maps names of associated items to the item's ID. Only includes items whose defining trait is
+    /// exported.
+    assoc_map: FxHashMap<String, SmallVec<[AssocItemId; 1]>>,
```

Comment:
Seems like it makes sense to use `SmalStr` here?  Or maybe just fold this into `fts` and use high-order bit of the value to distingusish ItemInNs from AssocItemId?

Reply:
I didn't put it in the fst because this is currently always queried exactly. I guess I could filter non-exact assoc. item matches out afterwards though.

Reply:
Yeah, it seems a bit wasteful to do heap allocation for each assoc item. It shouldn't matter that much (as assoc items are very are), but still

Reply:
(I went with SmolStr since the FST building code is already cursed enough)